soci/soci_index.go

```diff
@@ -297,29 +297,51 @@ func (b *IndexBuilder) Build(ctx context.Context, img images.Image) (*IndexWithM
 	}
 
 	sociLayersDesc := make([]*ocispec.Descriptor, len(manifest.Layers))
-	eg, ctx := errgroup.WithContext(ctx)
+	var wg sync.WaitGroup
+	errChan := make(chan error, len(manifest.Layers))
+
 	for i, l := range manifest.Layers {
-		i, l := i, l
-		eg.Go(func() error {
+		wg.Add(1)
+		go func(i int, l ocispec.Descriptor) {
+			defer wg.Done()
 			desc, err := b.buildSociLayer(ctx, l)
 			if err != nil {
-				return fmt.Errorf("could not build zTOC for layer %s: %w", l.Digest.String(), err)
+				if err != errUnsupportedLayerFormat {
+					errChan <- err
+				}
+				return
 			}
-			sociLayersDesc[i] = desc
-			return nil
-		})
+			if desc != nil {
+				// index layers must be in some deterministic order
+				// actual layer order used for historic consistency
+				sociLayersDesc[i] = desc
+			}
+		}(i, l)
 	}
-	if err := eg.Wait(); err != nil {
-		return nil, err
+
+	wg.Wait()
+
+errLoop:
```

Comment:
Before, if we got an error building a ztoc, we would exit with error and stop building the index. Now we just log the errors and continue. Shouldn't we still exit?